bloom-housing/bloom
# feat: record of listing copy

This PR addresses [#(806)](https://github.com/metrotranscom/doorway/issues/806)


## Description

Adds `copyOfId` field and `copyOf` relationship to the Listing table. This stores the id of the original listing into the duplicate.
This is surfaced in the listing export by adding two new columns: `Copy or Original` and `Copied From`.
Per product, `Copied From` should show the id.

## How Can This Be Tested/Reviewed?

Run partner portal and api and login as a admin user.
Have the listing id for the listing you would like to duplicate.
Fill out the payload for `listings/duplicate`. Set storedListing.id to the above.
Click the listing export button in the parter portal.
Open the listing csv and view/verify the new columns.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
